x-pack/plugins/reporting/public/components/screen_capture_panel_content.tsx

```diff
@@ -5,45 +5,56 @@
  * 2.0.
  */
 
-import { EuiSpacer, EuiSwitch, EuiSwitchEvent } from '@elastic/eui';
+import { EuiFormRow, EuiSwitch, EuiSwitchEvent } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
-import React, { Component, Fragment } from 'react';
+import moment from 'moment';
+import React, { Component } from 'react';
 import { ToastsSetup } from 'src/core/public';
-import { BaseParams } from '../../common/types';
+import { getDefaultLayoutSelectors } from '../../common';
+import { BaseParams, LayoutParams } from '../../common/types';
 import { ReportingAPIClient } from '../lib/reporting_api_client';
 import { ReportingPanelContent } from './reporting_panel_content';
 
+type LayoutID = 'canvas' | 'print';
+
 export interface Props {
   apiClient: ReportingAPIClient;
   toasts: ToastsSetup;
   reportType: string;
+  layoutOption?: LayoutID;
   objectId?: string;
   getJobParams: () => BaseParams;
+  requiresSavedState: boolean;
   isDirty?: boolean;
   onClose?: () => void;
 }
 
 interface State {
-  isPreserveLayoutSupported: boolean;
+  isPrintLayoutSupported: boolean;
   usePrintLayout: boolean;
+  isCanvasLayoutSupported: boolean;
+  useCanvasLayout: boolean;
 }
 
 export class ScreenCapturePanelContent extends Component<Props, State> {
   constructor(props: Props) {
     super(props);
 
-    const { objectType } = props.getJobParams();
-    const isPreserveLayoutSupported = props.reportType !== 'png' && objectType !== 'visualization';
```

Comment:
Removed this since elsewhere in the code it's fine/expected for PNG layout to provide "preserve" layout as the option. Therefore, preserve layout is always supported.